pkg/validation/internal/operatorhub.go

```diff
@@ -81,41 +101,202 @@ func validateBundleOperatorHub(bundle *manifests.Bundle) errors.ManifestResult {
+	// if the deprecated/unsupported APIs are found. See that the error only can be safely raised when the
+	// minKubeVersion value is >=1.22.
+	if !hasK8sVersion || semverK8sVersion.GE(k8s1_16) {
+		isUnsupported := semverK8sVersion.GE(k8s1_22) || semverMinKube.GE(k8s1_22)
+
+		// Note that if the operator bundle has both versions then, it will only raise a warning since it might
+		// can means that the operator has been covering both scenarios.
+		if hasDeprecatedAPIs(bundle) {
+			if isUnsupported && !hasV1(bundle) {
+				errs = append(errs, fmt.Errorf(apiDeprecationMsg))
+			} else {
+				warns = append(warns, fmt.Errorf(apiDeprecationMsg))
 			}
 		}
 	}
 
-	for _, link := range csv.Spec.Links {
-		if link.Name == "" || link.URL == "" {
-			errs = append(errs, fmt.Errorf("csv.Spec.Links elements should contain both name and url"))
+	return errs, warns
+}
+
+// hasDeprecatedAPIs returns true when the bundle has one CRD/Webhook using v1beta1
+func hasDeprecatedAPIs(bundle *manifests.Bundle) bool {
+	if len(bundle.V1beta1CRDs) > 0 {
+		return true
+	}
+	for _, obj := range bundle.CSV.Spec.WebhookDefinitions {
+		for _, v := range obj.AdmissionReviewVersions {
+			if v == "v1beta1" {
+				return true
+			}
 		}
-		if link.URL != "" {
-			_, err := url.ParseRequestURI(link.URL)
+	}
+	return false
+}
+
+// hasV1 returns true when the bundle has a CRD/Webhook using V1
+func hasV1(bundle *manifests.Bundle) bool {
+	if len(bundle.V1CRDs) > 0 {
+		return true
+	}
+	for _, obj := range bundle.CSV.Spec.WebhookDefinitions {
+		for _, v := range obj.AdmissionReviewVersions {
+			if v == "v1" {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+// validateHubCSVSpec will check the CSV against the criteria to publish an
+// operator bundle in the OperatorHub.io
+func validateHubCSVSpec(csv v1alpha1.ClusterServiceVersion) ([]error, []error) {
+	var errs []error
+	var warns []error
+
+	errs = checkSpecProviderName(csv, errs)
+	errs = checkSpecMaintainers(csv, errs)
+	errs = checkSpecLinks(csv, errs)
+	errs, warns = checkAnnotations(csv, errs, warns)
+	errs = checkSpecVersion(csv, errs)
+	errs, warns = checkSpecIcon(csv, errs, warns)
+	errs, warns = checkSpecMinKubeVersion(csv, warns, errs)
+	errs, warns = checkBundleName(csv, warns, errs)
+
+	return errs, warns
+}
+
+// checkBundleName will validate the operator bundle name informed via CSV.metadata.name.
+func checkBundleName(csv v1alpha1.ClusterServiceVersion, warns []error, errs []error) ([]error, []error) {
+	if len(strings.TrimSpace(csv.Name)) == 0 {
+		errs = append(errs, fmt.Errorf("csv.metadata.Name is empty. You should inform the operator bundle name e.g memcached-operator.v0.0.1"))
+	} else {
+		// The motivation for the following check is to ensure that operators authors knows that operator bundles names should
+		// follow a name convention with versioning
+		bundleNameSplit := strings.Split(csv.Name, ".v")
```

Comment:
Is `foo-operator.0.0.1` not allowed? `0.0.1` is stricter semver than `v0.0.1`.

Reply:
All unique value is allowed. We just raise the WARN here when it is not following the convention which is with `name.v<semver>`. Do you think that we should check if is `foo-operator.0.0.1` and not raise the WARN in this case?

Reply:
Any unique value is allowed. We just WARN the scenarios. 
If it has not 0.0.1 -> then is not following the versioning convention
if it is not operator-name.v<semver> -> is not following the naming convention

I improved that using regex as should be. Thank you.